src/sentry/incidents/action_handlers.py

```diff
@@ -48,11 +48,15 @@ def resolve(self, metric_value: int | float, new_status: IncidentStatus):
 
 class DefaultActionHandler(ActionHandler):
     def fire(self, metric_value: int | float, new_status: IncidentStatus):
-        if not RuleSnooze.objects.filter(alert_rule=self.incident.alert_rule).exists():
+        if not RuleSnooze.objects.filter(
+            alert_rule=self.incident.alert_rule, user_id__isnull=True
+        ).exists():
             self.send_alert(metric_value, new_status)
 
     def resolve(self, metric_value: int | float, new_status: IncidentStatus):
-        if not RuleSnooze.objects.filter(alert_rule=self.incident.alert_rule).exists():
+        if not RuleSnooze.objects.filter(
+            alert_rule=self.incident.alert_rule, user_id__isnull=True
+        ).exists():
```

Comment:
Not sure if we need to handle this in this pr, but I'm starting to wonder if we should have something on the manager for `RuleSnooze` that makes these queries for us.

Like `RuleSnooze.objects.disabled_for_team(alert_rule)` and `RuleSnooze.objects.disabled_for_user(alert_rule, user)` or something

Reply:
Yeah that could definitely be useful, we can do it in a follow up. I've been meaning to make a test fixture thing for muting an alert too.